src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDetectPreviewFeatureAnalyzer.cs

```diff
@@ -11,6 +12,227 @@ namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+                        {
+                            if (TryMatchGenericSyntaxNodeWithGivenSymbol(genericName, parameterSymbol, out SyntaxNode? previewNode))
+                            {
+                                return previewNode;
+                            }
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        // Handles both generic and non-generic return types
+        protected override SyntaxNode? GetPreviewReturnTypeSyntaxNodeForMethodOrProperty(ISymbol methodOrPropertySymbol, ISymbol previewReturnTypeSymbol)
+        {
+            ImmutableArray<SyntaxReference> methodOrPropertySymbolDeclaringReferences = methodOrPropertySymbol.DeclaringSyntaxReferences;
+
+            foreach (SyntaxReference? syntaxReference in methodOrPropertySymbolDeclaringReferences)
+            {
+                SyntaxNode methodOrPropertyDefinition = syntaxReference.GetSyntax();
+                if (methodOrPropertyDefinition is PropertyDeclarationSyntax propertyDeclaration)
+                {
+                    TypeSyntax returnType = propertyDeclaration.Type;
+                    returnType = GetElementTypeForNullableAndArrayTypeNodes(returnType);
+                    if (IsIdentifierNameSyntax(returnType, previewReturnTypeSymbol))
+                    {
+                        return returnType;
+                    }
+                    else if (returnType is GenericNameSyntax genericName)
+                    {
+                        if (TryMatchGenericSyntaxNodeWithGivenSymbol(genericName, previewReturnTypeSymbol, out SyntaxNode? previewNode))
+                        {
+                            return previewNode;
+                        }
+                    }
+                }
+                else if (methodOrPropertyDefinition is MethodDeclarationSyntax methodDeclaration)
+                {
+                    TypeSyntax returnType = methodDeclaration.ReturnType;
+                    returnType = GetElementTypeForNullableAndArrayTypeNodes(returnType);
+                    if (IsIdentifierNameSyntax(returnType, previewReturnTypeSymbol))
+                    {
+                        return returnType;
+                    }
+                    else if (returnType is GenericNameSyntax genericName)
+                    {
+                        if (TryMatchGenericSyntaxNodeWithGivenSymbol(genericName, previewReturnTypeSymbol, out SyntaxNode? previewNode))
+                        {
+                            return previewNode;
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private static bool TryMatchGenericSyntaxNodeWithGivenSymbol(GenericNameSyntax genericName, ISymbol previewReturnTypeSymbol, [NotNullWhen(true)] out SyntaxNode? syntaxNode)
+        {
+            if (IsSyntaxToken(genericName.Identifier, previewReturnTypeSymbol))
```

Comment:
@buyaa-n : This `if` block is the most meaningful change since you approved the PR. For a generic type like `AGenericType<int>`, `genericName.Identifier` returns `AGenericType`. `previewReturnTypeSymbol` here will also be `AGenericType<int>`, and `previewReturnTypeSymbol.Name` will return `AGenericType`, so this check will pass